app/src/googlePlay/java/com/smlnskgmail/jaman/hashchecker/MainActivity.java

```diff
@@ -63,6 +64,7 @@ public class MainActivity extends BaseActivity {
     protected void onCreate(@Nullable Bundle savedInstanceState) {
         App.appComponent.inject(this);
         super.onCreate(savedInstanceState);
+        MainActivitySingletone.setInstance(this);
```

Comment:
Why we need separated Activity here?

Reply:
It just save an instance of MainActivity to call .showFragment() funtion in WebLinksListAdapter.java.

Reply:
What can't we use activity instance from fragment?

Reply:
Activity and Singleton is not a best practice. If we can - we will find an other solution.

Reply:
Working on it

Reply:
> What can't we use activity instance from fragment?

`WebLinksListAdapter`  adapter is not aware of the `FragmentManager` of the hosting activity

Reply:
We can create a class which contains `private static Context context` and set it in `initPrivacyPolicy()` as `WebViewer.setContext(getActivity())` and will able to get showFragment() in WebLinksListAdapter.

Reply:
I will check this and give an answer as soon as possible. Not sure that I remember this thing correctly.

Reply:
@ManishDait, `WebLinksListAdapter` are using in `WebLinksBottomSheet`. You can add `Activity` parameter to `WebLinksListAdapter` and use it in `Holder` without calling singleton.

Reply:
@fartem thanks for helping. Resolve the singletone issue.